liquibase-core/src/main/java/liquibase/lockservice/StandardLockService.java

```diff
@@ -94,72 +97,96 @@ public void init() throws DatabaseException {
         boolean createdTable = false;
         Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc",  database);
 
-        if (!hasDatabaseChangeLogLockTable()) {
+        int maxIterations = 10;
+        for (int i = 0; i < maxIterations; i++) {
             try {
-                executor.comment("Create Database Lock Table");
-                executor.execute(new CreateDatabaseChangeLogLockTableStatement());
-                database.commit();
-                Scope.getCurrentScope().getLog(getClass()).fine(
-                        "Created database lock table with name: " +
-                                database.escapeTableName(
+                if (!hasDatabaseChangeLogLockTable(true)) {
+                    try {
+                        executor.comment("Create Database Lock Table");
+                        executor.execute(new CreateDatabaseChangeLogLockTableStatement());
+                        database.commit();
+                        Scope.getCurrentScope().getLog(getClass()).fine(
+                                "Created database lock table with name: " +
+                                        database.escapeTableName(
+                                                database.getLiquibaseCatalogName(),
+                                                database.getLiquibaseSchemaName(),
+                                                database.getDatabaseChangeLogLockTableName()
+                                        )
+                        );
+                        this.hasDatabaseChangeLogLockTable = true;
+                        createdTable = true;
+                        hasDatabaseChangeLogLockTable = true;
+                    } catch (DatabaseException e) {
+                        if ((e.getMessage() != null) && e.getMessage().contains("exist")) {
```

Comment:
Do we still need this special handling around "table already exists"? 

I think just the general handling of whatever errors get thrown should be the same, shouldn't it?

Reply:
This is a good question, I'll explore this further. I briefly tried removing this try/catch and it caused unsatisfactory results, but I suspect I may not have been doing a rollback in the higher try/catch.